### PR TITLE
Task5

### PR DESCRIPTION
**Pull request by generalov**
Frontend: https://dibzgqdmk7fo5.cloudfront.net (task4 PR: generalov/nodejs-aws-fe#4)
Implemented Endpoints: https://app.swaggerhub.com/apis-docs/e.generalov/nodejs-aws/1.4

## Prerequisites
---

- [x] The task is a continuation of Homework 4 and should be done in the same repos
- [x] Install the latest version of AWS SDK (https://aws.amazon.com/ru/sdk-for-node-js/)
- [x] Install the CSV parser package (https://www.npmjs.com/package/csv-parser)
- [x] Sign in into AWS Console and create a new S3 bucket with a folder, called uploaded
- [x] Pull latest changes from nodejs-aws-fe (https://github.com/rolling-scopes-school/nodejs-aws-fe)

## TASK 5.1
---

- [x] Create a new service, called **import-service**, with a its own **serverless.yml** file at the same level as product service.
- [x] Create a lambda function in that **serverless.yml** file, called **importProductsFile**, which will be triggered by the HTTP **GET** method.
  - [x] The requested URL should be **/import**. 
  - [x] The name of CSV file with products will be passed in a **query string** as a **name** parameter and should be described in the **serverless.yml** file as a **request parameter**.
  - [x]  A new **Signed URL** should be created with the following **Key: `uploaded/${fileName}`**.
  - [x] The response from the lambda should be the created **Signed URL**.
- [x] Update **serverless.yml** with policies to allow lambda functions to interact with **S3**.
- [x] The lambda endpoint should be integrated with the frontend by updating **import** property of the API paths configuration.

## TASK 5.2
---

- [x] Create a lambda function under the same **serverless.yml** file, called **importFileParser**, which will be triggered by the S3 **s3:ObjectCreated:*** event.
- [x] Configure the event to be triggered only by changes in the **uploaded** folder in S3.
- [x] The lambda function should use the **Readable stream** to get an object from S3, parse it via **csv-parser** and log each record to be shown in **CloudWatch**.
- [x] The response should be a correct HTTP response code.

## Evaluation criteria
---

Reviewers should verify the lambda functions by invoking them through provided URLs.
 
- [x] **1** - File **serverless.yml** contains configuration for **importProductsFile** function
- [x] **3** - The **importProductsFile** lambda function returns a correct response which can be used to upload a file into the **S3** bucket
- [x] **4** - Frontend application is integrated with **importProductsFile** lambda
- [x] **5** - The **importFileParser** lambda function is implemented and **serverless.yml** contains configuration for the lambda

## Additional (optional) tasks
---

- [x] **+1** - **async/await** is used in lambda functions
- [x] **+1** - **importProductsFile** lambda is covered by **unit** tests (**aws-sdk-mock** can be used to mock S3 methods - https://www.npmjs.com/package/aws-sdk-mock)
_I've tried to write tests using aws-sdk-mock but it incorrectly mocks the method `S3:getSignedUrlPromise()` returning a callbackish response `{ promise, createReadStream, ... }`. So Jest is being used_.
- [x] **+1** - At the end of the **stream** the lambda function should move the file from the **uploaded** folder into the **parsed** folder (move the file means that file should be copied into **parsed** folder, and then deleted from **uploaded** folder)

### CODE REVIEW

File: packages/import-service/.env [hunk 1]
```diff
@@ -1,4 +1,4 @@
 IMPORT_S3_BUCKET=nodejs-aws-task5-csv
```
Comment (Vladlan) [inline packages/import-service/.env:1]: Even this constants could be exposed, better to not commit .env file to a repo

**Review by Vladlan — COMMENTED** [review]
Not working csv file upload:
generalov_error

Discord Vladlan #8201

File: packages/import-service/.env (hunk 1 above)
Comment (generalov) [reply packages/import-service/.env:1]: They have a bit different rules:

> Note: .env, .env.development, and .env.production files should be included in your repository as they define defaults. .env*.local should be added to .gitignore, as those files are intended to be ignored. .env.local is where secrets can be stored.

https://www.serverless.com/plugins/serverless-dotenv-plugin

So, I'm storing confidential data in .env.*.local files.

**Comment by generalov** [comment]
> Not working csv file upload:
> generalov_error
> 
> Discord Vladlan #8201

@Vladlan Sorry, I forget to deploy the updated frontend. Could you please check it one more time?

File: packages/import-service/.env (hunk 1 above)
Comment (Vladlan) [reply packages/import-service/.env:1]: Ok

**Review by Vladlan — APPROVED** [review]
LGTM. Final mark 8/8.

**Pull request merged**